### ISSUE TITLE
Issue with validation codes generation

### ISSUE THREAD

**Issue by TheSuperiorStanislav** [ghazi-git/drf-standardized-errors#26]
Hi, I discovered an issue when testing via https://github.com/snok/drf-openapi-tester

I have a related field in serializer and also have validation of it in `validate_field_name`. the problem is that in spec there are only supported codes `does_not_exist`, `incorrect_type`, `null`, and `required`.  There is no `invalid` code.
An example of a model and serializer is below

```python
class Recommendation(BaseModel):
    """Model defines data for a recommendation from MAL.

    Recommendation works in the following way: If you `liked` this anime, you would
    like `recommend`ed anime.

    """

    mal_id = models.CharField(
        max_length=255,
        unique=True,
        null=True,
        blank=True,
        editable=False,
        verbose_name=_(
            "Constructed Id of recommendation from MAL",
        ),
        help_text=_(
            "MAL Id constructed of three parts: "
            "username who gave the recommendation, "
            "mal_id of anime which you liked,"
            "mal_id of anime which user recommends you to watch",
        ),
    )

    liked = models.ForeignKey(
        to="anime.Anime",
        on_delete=models.CASCADE,
        related_name="recommendation_liked",
        verbose_name=_("Liked anime"),
    )

    recommend = models.ForeignKey(
        to="anime.Anime",
        on_delete=models.CASCADE,
        related_name="recommendation_recommend",
        verbose_name=_("Recommended anime"),
    )

    reason = models.TextField(
        verbose_name=_("Reason"),
    )

    recommended_by = models.ForeignKey(
        null=True,
        to="users.User",
        on_delete=models.CASCADE,
        related_name="recommendations",
        verbose_name=_("Recommended by"),
    )

class RecommendationSerializer(ModelBaseSerializer):
    """Serializer to show info about recommendation."""

    class Meta:
        model = models.Recommendation
        fields = (
            "id",
            "created",
            "modified",
            "reason",
            "liked",
            "recommend",
        )

    def validate_recommend(self, data):
            raise ValidationError(
                _("Some error"),
            )
```

**Comment by ghazi-git** [comment]
Actually, error codes are collected from the error messages defined for each field rather than actually inspecting the code and identifying each `raise ValidationError`. In order to see the `invalid` error code in the generated schema, you'll need to follow the [example from the docs](https://drf-standardized-errors.readthedocs.io/en/latest/openapi.html#notes) (last item under "Notes").

Another option which will be available starting next release (0.12.5) is using the `extend_validation_errors` decorator, sth like
```python
@extend_validation_errors(["invalid"], field_name="recommend", method="post")
```
The error code will be added to the list of already collected error codes for validation errors. [More info in the corresponding section in the docs](https://drf-standardized-errors.readthedocs.io/en/latest/openapi.html#customize-error-codes-on-an-operation-basis)

**Comment by TheSuperiorStanislav** [comment]
@ghazi-git With 0.12.5 it's working as expected, thank you
